src/train/dataset.py

```diff
@@ -66,111 +85,326 @@ def preprocess(x: pl.DataFrame):
+    """DatasetLoader for Training."""
+
+    def __init__(self, base_path: str = "", max_rows: int = -1) -> None:
+        """Initialise data sets.
+
+        Args:
+            base_path (str, optional): Base path to data set folder. Defaults to "".
+            max_rows (int, optional): Maximum rows to consider. Defaults to -1.
+        """
+        logger.info("Initialise DatasetLoader")
+        self.base_path = base_path
+        self.max_rows = max_rows
+        logger.info("Finished initialisation.")
+
+    @property
+    def dgta_dataset(self) -> Dataset:
         self.dgta_data = Dataset(
-            data_path="/home/smachmeier/projects/heiDGA/data/dgta/dgta-benchmark.parquet",
+            data_path=f"{self.base_path}/dgta/dgta-benchmark.parquet",
             cast_dataset=cast_dgta,
+            max_rows=self.max_rows,
         )
+        return self.dgta_data
 
+    @property
+    def dga_dataset(self) -> Dataset:
         self.dga_data = Dataset(
-            data_path="/home/smachmeier/projects/heiDGA/data/360_dga_domain.csv",
+            data_path=f"{self.base_path}/360_dga_domain.csv",
             cast_dataset=cast_dga,
+            max_rows=self.max_rows,
         )
+        return self.dga_data
 
+    @property
+    def bambenek_dataset(self) -> Dataset:
         self.bambenek_data = Dataset(
-            data_path="/home/smachmeier/projects/heiDGA/data/bambenek_dga_domain.csv",
+            data_path=f"{self.base_path}/bambenek_dga_domain.csv",
             cast_dataset=cast_bambenek,
+            max_rows=self.max_rows,
         )
+        return self.bambenek_data
 
+    @property
+    def cic_dataset(self) -> Dataset:
         self.cic_data = Dataset(
             data_path=[
-                "/home/smachmeier/projects/heiDGA/example/CICBellDNS2021_CSV_benign.csv",
-                "/home/smachmeier/projects/heiDGA/example/CICBellDNS2021_CSV_malware.csv",
-                "/home/smachmeier/projects/heiDGA/example/CICBellDNS2021_CSV_phishing.csv",
-                "/home/smachmeier/projects/heiDGA/example/CICBellDNS2021_CSV_spam.csv",
+                f"{self.base_path}/cic/CICBellDNS2021_CSV_benign.csv",
+                f"{self.base_path}/cic/CICBellDNS2021_CSV_malware.csv",
+                f"{self.base_path}/cic/CICBellDNS2021_CSV_phishing.csv",
+                f"{self.base_path}/cic/CICBellDNS2021_CSV_spam.csv",
             ],
             cast_dataset=cast_cic,
+            max_rows=self.max_rows,
         )
+        return self.cic_data
 
     @property
-    def dgta_dataset(self) -> Dataset:
-        return self.dgta_data
-
-    @property
-    def dga_dataset(self) -> Dataset:
-        return self.dga_data
-
-    @property
-    def bambenek_dataset(self) -> Dataset:
-        return self.bambenek_data
-
-    @property
-    def cic_dataset(self) -> Dataset:
-        return self.cic_data
+    def dgarchive_dataset(self) -> Dataset:
+        self.dgarchive_data = Dataset(
+            data_path=[
```

Comment:
Maybe we should move the paths to their own file to simplify and shorten the code